src/generator/generator_visitor.ts

```diff
@@ -328,34 +328,38 @@ class GeneratorVisitor extends Visitor {
         const value = this.popExpression();
 
         if (isIdentifier(val)) {
-            this.statements.push(this.module.set_local(this.getVariableIndex(val.name), value));
+            this.statements.push(this.setLocal(val, value));
         } else if (isMemberExpression(val)) {
-            this.setArrayElement(val, value);
+            this.statements.push(this.setArrayElement(val, value));
         } else {
             throw new Error('Assignment to non-identifier or member expression');
         }
     }
 
+    private setLocal(val: Identifier, value: Expression) {
+        return this.module.set_local(this.getVariableIndex(val.name), value);
+    }
+
     private getArrayElement(node: MemberExpression) {
         super.visitMemberExpression(node);
 
         // The offset can not be used, because the memberaccess value can also be a mathematical term or a variable
-        this.expressions.push(this.module.i32.load(0, 4, this.getPointer()));
```

Comment:
It's because of the methods name. When calling a `get...()` method I expect a value to be returned. Also we take control out of this helper method, which increases reusability.